boards: arm: arduino_nano_33_ble: Add i2c sensors to DTS

benjaminbjornsson opened the pull request:
This commit adds the hts221, lps22hb and apds9960 sensors to
the DTS since these sensors are present on the Sense version
of the board.

These changes were verified using the sample of each sensor.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

gmarull commented on boards/arm/arduino_nano_33_ble/arduino_nano_33_ble.dts:
```diff
@@ -95,6 +95,28 @@
+
+	hts221: hts221@5f {
+		compatible = "st,hts221";
+		status = "disabled";
+		reg = <0x5f>;
+		label = "HTS221";
+	};
+
+	lps22hb: lps22hb-press@5c {
+		compatible = "st,lps22hb-press";
+		status = "disabled";
+		reg = <0x5c>;
+		label = "LPS22HB";
+	};
+
+	apds9960: apds9960@39 {
+		compatible = "avago,apds9960";
+		status = "disabled";
```
should these be `status = "okay"`? They won't get enabled unless `CONFIG_SENSOR=y` and `CONFIG_XXX=y` (XXX=sensor driver)

benjaminbjornsson replied:
Sounds resonable, thank you 👍

MaureenHelm commented:
Please rebase

benjaminbjornsson committed:
boards: arm: arduino_nano_33_ble: Add uart console to board defconfig

This commit adds uart console to the board defconfig.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson commented:
Should I remove the regulator-fixed-sync compatible since it expects no delay? I had to add the startup delay to the vdd_env node to make it work. Haven't been able to verify it myself but this post suggests that the reason for the delay is to make some time for the vdd_env pin to go high: https://devzone.nordicsemi.com/f/nordic-q-a/77845/problem-initializing-i2c-on-arduino-nano-33-ble.

gmarull commented on boards/arm/arduino_nano_33_ble/arduino_nano_33_ble.dts:
```diff
@@ -49,6 +49,7 @@
 		regulator-name = "vdd_env";
 		enable-gpios = <&gpio0 22 GPIO_ACTIVE_HIGH>;
 		regulator-boot-on;
+		startup-delay-us = <5000>;
```
why this change? it's not described in the commit message.

gmarull reviewed (changes requested):
you mention that added sensors are only available on the sense version of the board. where is this board variant defined? Please check other boards that do have variants, e.g. longan_nano.

benjaminbjornsson committed:
boards: arm: arduino_nano_33_ble: Add support for sense version of board

This commit adds support for the Arduino Nano 33 BLE Sense board.

The DTS is separated into a common dtsi and one DTS for each version
of the board. The sensors with existing drivers: hts221, lps22hb and
apds9960 are added to the DTS of the Sense version.

A startup delay is added to the vdd_env node. This is done to give some
time for the sensors VDD/VCC to go high before the sensor drivers are
initialized.

The functionality was verified using the driver sample of each sensor.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>
